151629713 create api users

aenkya opened the pull request:
### What does this PR do?
Allow applications to register users via an API endpoint

### Description of task to be completed?
It should be possible to create new users via the API. External apps might need to register accounts on behalf of their users without having to send them over to the wger instance

### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2116052/stories/151629713

Commit:
Tests for Registration of Users via API

Commit:
(feat 151629713) Create API Users Model
A model that maps the user to the application of origin

Commit:
(feat 151629713) Generate Object Serializers
- UserSerializer
- ApiUserSerializer

Commit:
(feat 151629713) Add API Endpoints
- ApiUsersView for viewing users registered via API
- UserRegistrationView for creating new users via the API

Commit:
(feat 151629713) Create Users via API
Modified tests to use api url instead of reverse
Modified create function to pick api_key from request data

Commit:
(feat 151629713): Add migrations configurations

Commit:
(feat 151629713) Remove Guardian Permissions

Commit:
(feat 151629713) Modify Makemigrations

Commit:
(feat 151629713) Modify Makemigrations

Commit:
Merge branch '151629713-create-api-users' of https://github.com/andela/wger-shockwave into 151629713-create-api-users

Commit:
(feat 151629713): Modify requirements.txt

emugaya commented on wger/core/api/views.py:
```diff
@@ -108,3 +109,50 @@ class WeightUnitViewSet(viewsets.ReadOnlyModelViewSet):
+        # Create APIUser object with Application and created user
+        ApiUsers.objects.create(app=cuser, app_user=created_user)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```
This line doesn't meet pep8 guidelines.

aenkya replied:
Alright thanks

Commit:
(feat 151629713) PEP8 linting

Commit:
(feat 151629713) PEP8 linting

Commit:
Merge branch '151629713-create-api-users' of https://github.com/andela/wger-shockwave into 151629713-create-api-users

emugaya commented:
LGTM

emugaya commented on wger/core/api/views.py:
```diff
@@ -108,3 +110,53 @@ class WeightUnitViewSet(viewsets.ReadOnlyModelViewSet):
+            return ApiUserSerializer
+        return UserSerializer
+
+    def create(self, request):
```
Your creating the user before verifying whether the account/ app that is creating the account is allowed to do so. User descriptive names for your variables. Consider breaking these functions into two different methods(Create user, and Create API user)

emugaya commented on wger/urls.py:
```diff
@@ -118,6 +118,11 @@
     r'setting-weightunit',
     core_api_views.WeightUnitViewSet,
     base_name='setting-weight-unit')
+router.register(
```
I have tested these two endpoints and they give errors:
`http://localhost:8000/api/v2/user/`
![image](https://user-images.githubusercontent.com/8415776/31962720-038eea7a-b907-11e7-81b3-0d2016fe5496.png)
`http://localhost:8000/api/v2/apiusers/`
![image](https://user-images.githubusercontent.com/8415776/31962788-30ea874a-b907-11e7-9544-923b8ac69634.png)

Make sure these endpoints can onkly be accessed by only authenticated users

Commit:
(feat 151629713): Add API Users
Modify logic to prevent creation of user before verifying the application making the requests
